rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/RMRest.java

```diff
@@ -487,6 +514,61 @@ public int getNodeSourcePingFrequency(String sessionId, String sourceName)
         return orThrowRpe(rm.getNodeSourcePingFrequency(sourceName).getIntValue());
     }
 
+    @Override
+    public Set<String> acquireNodes(String sessionId, String sourceName, int numberNodes, boolean synchronous,
+            int timeout, String nodeConfigJson) throws NotConnectedException, RestException {
+        if (numberNodes <= 0) {
+            throw new IllegalArgumentException("invalid numberNodes: " + numberNodes);
+        }
+        ResourceManager rm = checkAccess(sessionId);
+        ObjectMapper mapper = new ObjectMapper();
+        Map<String, Object> nodeConfig;
+        try {
+            nodeConfig = mapper.readValue(nodeConfigJson, new TypeReference<Map<String, ?>>() {
+            });
+        } catch (IOException e) {
```

Comment:
are you sure only IOException can be thrown ? What if the json is not valid. Is it not better to catch Exception ?